cli/support.go

```diff
@@ -0,0 +1,202 @@
+package cli
+
+import (
+	"archive/zip"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/sloghuman"
+	"github.com/coder/coder/v2/cli/clibase"
+	"github.com/coder/coder/v2/coderd/util/tz"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/support"
+)
+
+func (r *RootCmd) support() *clibase.Cmd {
+	var outputPath string
+	client := new(codersdk.Client)
+	cmd := &clibase.Cmd{
+		Use:   "support <workspace> [<agent>]",
+		Short: "Generate a support bundle to troubleshoot issues.",
+		Long:  `This command generates a file containing detailed troubleshooting information about the Coder deployment and workspace connections. You must specify a single workspace (and optionally an agent name).`,
+		Middleware: clibase.Chain(
+			clibase.RequireRangeArgs(0, 2),
+			r.InitClient(client),
+		),
+		Handler: func(inv *clibase.Invocation) error {
+			var (
+				log = slog.Make(sloghuman.Sink(inv.Stderr)).
+					Leveled(slog.LevelDebug)
+				deps = support.Deps{
+					Client: client,
+					Log:    log,
+				}
+			)
+
+			if len(inv.Args) == 0 {
+				return xerrors.Errorf("must specify workspace name")
+			}
+			ws, err := namedWorkspace(inv.Context(), client, inv.Args[0])
+			if err != nil {
+				return err
+			}
+
+			deps.WorkspaceID = ws.ID
+
+			agentName := ""
+			if len(inv.Args) > 1 {
+				agentName = inv.Args[1]
+			}
+
+			agt, found := findAgent(agentName, ws.LatestBuild.Resources)
+			if !found {
+				return xerrors.Errorf("could not find agent named %q for workspace", agentName)
+			}
+
+			deps.AgentID = agt.ID
+
+			bun, err := support.Run(inv.Context(), &deps)
+			if err != nil {
+				return err
+			}
+
+			if outputPath == "" {
+				cwd, err := filepath.Abs(".")
+				if err != nil {
+					return xerrors.Errorf("could not determine current working directory: %w", err)
+				}
+				loc, err := tz.TimezoneIANA()
+				if err != nil {
+					loc = time.UTC
+				}
+				tsStr := time.Now().In(loc).Format("2006-01-02-150405")
+				fname := "coder-support-" + tsStr + ".zip"
+				outputPath = filepath.Join(cwd, fname)
+			}
+
+			w, err := os.Create(outputPath)
+			if err != nil {
+				return xerrors.Errorf("create output file: %w", err)
+			}
+			zwr := zip.NewWriter(w)
+			defer zwr.Close()
+			if err := writeBundle(bun, zwr); err != nil {
+				return xerrors.Errorf("write support bundle to %s: %w", outputPath, err)
+			}
+			return nil
+		},
+		Hidden: true, // TODO: un-hide
+	}
+	cmd.Options = clibase.OptionSet{
+		{
+			Flag:          "output",
+			FlagShorthand: "o",
+			Env:           "CODER_SUPPORT_BUNDLE_OUTPUT",
+			Description:   "Path to which to output the generated support bundle. Defaults to coder-support-YYYYmmdd-HHMMSS.zip.",
```

Comment:
```suggestion
			Description:   "File path for writing the generated support bundle. Defaults to coder-support-YYYYmmdd-HHMMSS.zip.",
```
Path could be interpreted as folder, I feel.